java-frontend/src/test/java/org/sonar/java/se/Expectations.java

```diff
@@ -198,31 +198,39 @@ public String toString() {
   }
 
 
-  Optional<String> containFlow(List<AnalyzerMessage> flow) {
-    if (flowLines == null) {
-      flowLines = computeFlowLines();
+  Optional<String> containFlow(List<AnalyzerMessage> actual) {
+    List<Integer> actualLines = flowToLines(actual, AnalyzerMessage::getLine);
+    List<Map.Entry<String, SortedSet<FlowComment>>> expectedFlows = Multimaps.asMap(flows).entrySet().stream()
```

Comment:
That's ugly (and a huge pain to follow). What about a dedicated type to encode that?

Reply:
I don't want to introduce dedicated type, but I simplified this a lot, it was unnecessarily complex